airflow/include/tasks/extract/utils/weaviate/ask_astro_weaviate_hook.py

```diff
@@ -392,28 +423,49 @@ def ingest_data(
 
         self.logger.info(f"Passing {len(df)} objects for ingest.")
 
-        batch_errors = self.batch_ingest(
+        self.batch_ingest(
             df=df,
             class_name=class_name,
             uuid_column=uuid_column,
             vector_column=vector_column,
             batch_params=batch_params,
             existing=existing,
             verbose=verbose,
+            tenant=tenant,
         )
 
-        if existing == "upsert" and batch_errors:
-            self.logger.warning("Error during upsert. Rolling back all inserts for docs with errors.")
-            rollback_errors = self.handle_upsert_rollback(
-                objects_to_upsert=objects_to_upsert, batch_errors=batch_errors, class_name=class_name, verbose=verbose
-            )
+        if existing == "upsert":
+            if len(self.batch_errors) > 0:
+                self.logger.warning("Error during upsert. Rolling back all inserts for docs with errors.")
+                rollback_errors, objects_to_remove = self.handle_upsert_rollback(
+                    objects_to_upsert=objects_to_upsert, class_name=class_name, verbose=verbose
+                )
+
+                deletion_errors = self.handle_successful_upsert(
+                    objects_to_remove=objects_to_remove, class_name=class_name, verbose=verbose
+                )
 
-            if len(rollback_errors) > 0:
-                self.logger.error("Errors encountered during rollback.")
-                raise AirflowException("Errors encountered during rollback.")
+                rollback_errors += deletion_errors
+
+                if len(rollback_errors) > 0:
+                    self.logger.error("Errors encountered during rollback.")
+                    [self.logger.error(f"{rollback_error}" for rollback_error in rollback_errors)]
```

Comment:
```suggestion
                    self.logger.error("\n".join(rollback_errors))
```

not sure whether it makes more sense